### DISCUSSION TITLE
Action request: Create/manage Fastmail masked email addresses

### DISCUSSION THREAD

**Discussion by sentience** [Ptujec/LaunchBar#35] [Ideas]
Fastmail lets you create [masked email addresses](https://www.fastmail.help/hc/en-us/articles/4406536368911-Masked-Email) that you can use to register or subscribe to individual services, and then cancel them when you no longer need them – or discover who leaked your address when you receive unexpected email!

There is an [npm package](https://www.npmjs.com/package/@lusc/fastmail-masked-email) that uses Fastmail's public API to create and manage masked emails.

It would be great to be able to use Launchbar to create and enter a new masked email as needed on a website, or anywhere else. It would also be cool to be able to browse/search my existing masked emails, and even delete those I don't need anymore.

**Comment by Ptujec** [comment]
I don't use Fastmail, but it sounds like an interesting idea. Right now I am pretty busy and it looks a bit intimidating …

**Comment by prenagha** [comment]
I use this Secret Inbox app on Mac and iPhone and it works pretty well. You can setup a hotkey to create a new masked email and put in on the clipboard.

https://apps.apple.com/us/app/secret-inbox/id6462335670

**Comment by Ptujec** [comment]
> I use this Secret Inbox app on Mac and iPhone and it works pretty well. You can setup a hotkey to create a new masked email and put in on the clipboard.
> 
> https://apps.apple.com/us/app/secret-inbox/id6462335670

I looks like the app has shortcut support. With this and a simple line of applescript it should be possible to enter a new masked email via LaunchBar. It would at least be a good start.
![CleanShot 2024-06-06 at 12 51 07](https://github.com/Ptujec/LaunchBar/assets/617932/00116274-da0f-4a28-9b45-b8e831975322)

**Comment by mlinzner** [comment]
Quick proof-of-concept. Works fine, but there's a lot more that could be done: 

https://gist.github.com/mlinzner/35299df24e09f8177944a5df18dbf55e
